File: smash-ranks-android/app/src/main/java/com/garpr/android/adapters/TournamentsAdapter.java
package com.garpr.android.adapters;

import android.content.Context;
import android.support.annotation.LayoutRes;
import android.support.annotation.NonNull;
import android.view.LayoutInflater;

import com.garpr.android.R;
import com.garpr.android.models.AbsTournament;

public class TournamentsAdapter extends BaseAdapter<AbsTournament> {

    public TournamentsAdapter(@NonNull final Context context) {
        super(context);
        setHasStableIds(true);
    }

    public TournamentsAdapter(@NonNull final LayoutInflater layoutInflater) {
        super(layoutInflater);
        setHasStableIds(true);
    }

    @Override
    public long getItemId(final int position) {
        return getItem(position).hashCode();
    }

    @LayoutRes
    @Override
    public int getItemViewType(final int position) {
        return R.layout.item_tournament;
    }

}

File: smash-ranks-android/app/src/main/java/com/garpr/android/adapters/TournamentsAdapter.kt
package com.garpr.android.adapters

import androidx.annotation.LayoutRes
import com.garpr.android.R
import com.garpr.android.data.models.AbsTournament

class TournamentsAdapter : BaseAdapter<AbsTournament>() {

    init {
        setHasStableIds(true)
    }

    override fun getItemId(position: Int): Long {
        return getItem(position).hashCode().toLong()
    }

    @LayoutRes
    override fun getItemViewType(position: Int) = R.layout.item_tournament

}
